### PR TITLE
Factoring out a hard-coded contract template library.

### PR DESCRIPTION
**Pull request by merivale**
When we have a proper library/marketplace for Marlowe contracts, we will want some kind of database. For now, we just need a few of our own examples, so the simplest thing is just to write them in directly as PureScript.

This PR also removes the contract name from the metadata, and adds a contract type - with a function for mapping these contract types to a full name.

While I was at it, I also moved the dashboard template library display slightly closer to the design target:

Screenshot 2021-03-05 at 13 39 03

### CODE REVIEW

**Review by palas — APPROVED** [review]
I would keep the contract name so that we can have several templates for each type (eventually). But other than that it looks good 👍

File: web-common-marlowe/src/Marlowe/Extended.purs [hunk 1]
```diff
@@ -29,15 +29,45 @@ type ContractTemplate
...
+  = { contractType :: ContractType
     , contractDescription :: String
     , roleDescriptions :: Map S.TokenName String
     , slotParameterDescriptions :: Map String String
     , valueParameterDescriptions :: Map String String
     , choiceDescriptions :: Map String String
     }
 
+data ContractType
+  = ES
```
Comment (hrajchert) [inline web-common-marlowe/src/Marlowe/Extended.purs:41]: I'd favour long names here. We are going to use this in code, so having ES, EC and having to remember which one was which is painful. I'd rename to 

```
data ContractType 
  = Escrow 
  | EscrowWithCollatoral
  | ...
```

File: web-common-marlowe/src/Marlowe/Extended.purs [hunk 2]
```diff
@@ -29,15 +29,45 @@ type ContractTemplate
...
+  | EC
+  | ZC
+  | CB
+  | S
+  | CD
+
+derive instance eqContractType :: Eq ContractType
+
+instance showContractType :: Show ContractType where
```
Comment (hrajchert) [inline web-common-marlowe/src/Marlowe/Extended.purs:50]: We add more meaning to the `Show` typeclass than we should. I'd use a `contractAcronym` function

**Review by hrajchert — APPROVED** [review]
LGTM... minor change

**Pull request merged**